### ISSUE TITLE
Incorrect Noise Period value passed to hardware

### ISSUE THREAD

**Issue by howprice** [remy/next-ayfx#1]
src/ayfx.asm Line 312 should be `cp #20`. As it stands `sub #20`, subtracts from the Noise Period value from the accumulator to perform the end-of-effect test, but the original value is not restored before storing in Line 322 `ld (afxNseMix+1),a` which is later used as the value to write directly into the AY-3 Noise Period register.

The fix to Line 312 requires that Line 313 be changed to `jr nz,afxFrame2`

This is a bug in the original code.

**Comment by remy** [comment]
Thank you for reporting this. I read the message at totally the wrong time and promptly forgot about it!

I've got this PR ready to merge: https://github.com/remy/next-ayfx/pull/2/files - I know it's a small change, but if you could confirm it I'd appreciate it.

Thank you again.

**Comment by howprice** [comment]
Hi. Doesn't the `sub `have to become a `cp `to avoid incorrectly altering the value?

**Comment by howprice** [comment]
I think I have approved the change. Thanks!

**Comment by howprice** [comment]
I've realised that this fix may not have been required. AY reg 6 takes a 5 bit value for the noise period, so subtracting 20h from the value will not affect the lowest 5 bits. Assuming the chip ignores the top 3 bits the code was probably fine as it was.

The argument for keeping the change is that the code is now more correct with respect to the AYFX spec, which says that the end of the effect should end with a change of noise period of 20h. Before the change a greater or equal to 20h (any out of range value) would indicate the end of the effect.

I suppose there is a chance that some data somewhere relies on the old behavior. Sorry if I've wasted your time.
